modules/System Admin/file_upload.php

```diff
@@ -0,0 +1,133 @@
+
+        $row = $form->addRow()->addClass('useSeparator');
+            $row->addLabel('fileSeparator', __('Separator Character'));
+            $row->addTextField('fileSeparator')->required()->maxLength(1);
+
+        $row = $form->addRow()->addClass('useSeparator');
+            $row->addLabel('fileSection', __('Section Number'))->description(__('Once split, which section contains the username? Counting from 1.'));
+            $row->addNumber('fileSection')->required()->maxLength(1);
+
+        $form->addRow()->addHeading('Upload Photos & Files', __('Upload Photos & Files'));
+
+        $types = [
+            'userPhotos'        => __('User Photos'),
+            'personalDocuments' => __('Personal Documents'),
+            'customFields'      => __('Custom Fields'),
+        ];
+
+        $row = $form->addRow();
+            $row->addLabel('type', __('Type'));
+            $row->addSelect('type')->fromArray($types)->placeholder()->required();
+
+        $form->toggleVisibilityByClass('personalDocuments')->onSelect('type')->when('personalDocuments');
+        $personalDocumentTypes = $container->get(PersonalDocumentTypeGateway::class)->selectDocumentTypesWithFileUpload()->fetchKeyPair();
+        $row = $form->addRow()->addClass('personalDocuments');
+            $row->addLabel('gibbonPersonalDocumentTypeID', __('Personal Document'));
+            $row->addSelect('gibbonPersonalDocumentTypeID')->fromArray($personalDocumentTypes)->required();
+
+        $form->toggleVisibilityByClass('customFields')->onSelect('type')->when('customFields');
+        $customFields = $container->get(CustomFieldGateway::class)->selectCustomFieldsWithFileUpload();
+        $row = $form->addRow()->addClass('customFields');
+            $row->addLabel('gibbonCustomFieldID', __('Personal Document'));
+            $row->addSelect('gibbonCustomFieldID')->fromResults($customFields, 'groupBy')->required();
+
+        $form->toggleVisibilityByClass('userPhotos')->onSelect('type')->when('userPhotos');
+
+        $form->addRow()->addClass('userPhotos')->addContent(Format::alert(__('User photos will automatically be scaled down to a maximum of 360px by 480px and cropped to an aspect ratio of 1.2. You can optionally adjust the zoom and focal point of the cropped images below.'), 'message'));
+
+        $row = $form->addRow()->addClass('userPhotos');
+            $row->addLabel('zoom', __('Zoom Rate'));
+            $row->addRange('zoom', 100, 150, 1)->setValue(100);
+
+        $row = $form->addRow()->addClass('userPhotos');
+            $row->addLabel('focalX', __('Horizontal Focal Point'));
+            $row->addRange('focalX', 0, 100, 5)->setValue(50);
+
+        $row = $form->addRow()->addClass('userPhotos');
+            $row->addLabel('focalY', __('Vertical Focal Point'));
+            $row->addRange('focalY', 0, 100, 5)->setValue(50);
+
+        $row = $form->addRow();
+            $row->addFooter();
+            $row->addSubmit();
+
+    } else if ($step == 3) {
```

Comment:
Where did $step == 2 go?!

Reply:
Well spotted. Step 2 needed to be on a hybrid process/display page, in order to peek into the contents of the zip file and display the matches. Its enough of a chunk of code that I still wanted it in it's own page, rather than crammed into the initial form.